In [2]:
import pandas as pd
import scipy.stats as stats
import matplotlib.pyplot as plt
import seaborn as sns
db=pd.read_csv('Sales_data_zone_wise.csv')

In [3]:
db.head()

Month  Zone - A  Zone - B  Zone - C  Zone - D
0  Month - 1   1483525   1748451   1523308   2267260
1  Month - 2   1238428   1707421   2212113   1994341
2  Month - 3   1860771   2091194   1282374   1241600
3  Month - 4   1871571   1759617   2290580   2252681
4  Month - 5   1244922   1606010   1818334   1326062

In [4]:
db.tail()

Month  Zone - A  Zone - B  Zone - C  Zone - D
24  Month - 25   1256333   1622671   1521792   1695122
25  Month - 26   1422853   1715465   1853636   1520406
26  Month - 27   1384426   1983163   1611169   1289160
27  Month - 28   1616640   1547991   2128022   2178267
28  Month - 29   1310654   1660092   1947119   1854412

In [5]:
db.isnull().sum()

Month       0
Zone - A    0
Zone - B    0
Zone - C    0
Zone - D    0
dtype: int64

In [6]:
db.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29 entries, 0 to 28
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   Month     29 non-null     object
 1   Zone - A  29 non-null     int64 
 2   Zone - B  29 non-null     int64 
 3   Zone - C  29 non-null     int64 
 4   Zone - D  29 non-null     int64 
dtypes: int64(4), object(1)
memory usage: 1.3+ KB


In [7]:
db.describe()

Zone - A      Zone - B      Zone - C      Zone - D
count  2.900000e+01  2.900000e+01  2.900000e+01  2.900000e+01
mean   1.540493e+06  1.755560e+06  1.772871e+06  1.842927e+06
std    2.619401e+05  1.683899e+05  3.331937e+05  3.750165e+05
min    1.128185e+06  1.527574e+06  1.237722e+06  1.234311e+06
25%    1.305972e+06  1.606010e+06  1.523308e+06  1.520406e+06
50%    1.534390e+06  1.740365e+06  1.767047e+06  1.854412e+06
75%    1.820196e+06  1.875658e+06  2.098463e+06  2.180416e+06
max    2.004480e+06  2.091194e+06  2.290580e+06  2.364132e+06

In [11]:
print('There are ',db.shape[0],'rows and ',db.shape[1],'columns in the dataset.')

There are  29 rows and  5 columns in the dataset.


# 1. The mean sales generated by each zone

In [10]:
import warnings
warnings.filterwarnings('ignore')
db.mean()

Zone - A    1.540493e+06
Zone - B    1.755560e+06
Zone - C    1.772871e+06
Zone - D    1.842927e+06
dtype: float64

# 2.Total sales generated by all zones for each month

In [12]:
column_list = list(db)
column_list.remove("Month")
da=db
da["Total sales per month of all zones"]= db[column_list].sum(axis=1)
da

Month  Zone - A  Zone - B  Zone - C  Zone - D  \
0    Month - 1   1483525   1748451   1523308   2267260   
1    Month - 2   1238428   1707421   2212113   1994341   
2    Month - 3   1860771   2091194   1282374   1241600   
3    Month - 4   1871571   1759617   2290580   2252681   
4    Month - 5   1244922   1606010   1818334   1326062   
5    Month - 6   1534390   1573128   1751825   2292044   
6    Month - 7   1820196   1992031   1786826   1688055   
7    Month - 8   1625696   1665534   2161754   2363315   
8    Month - 9   1652644   1873402   1755290   1422059   
9   Month - 10   1852450   1913059   1754314   1608387   
10  Month - 11   2004480   1740365   1317869   1970069   
11  Month - 12   1537539   1875658   1237722   1460165   
12  Month - 13   1254939   1588473   1348629   1733383   
13  Month - 14   1821799   1844081   1617376   1872259   
14  Month - 15   1128185   1804613   1767047   1234311   
15  Month - 16   1491534   1542053   1882159   1590913   
16  Month - 17   1182110   1552388   2234469   2180416   
17  Month - 18   1770324   1699213   1249821   2364132   
18  Month - 19   1181635   1959268   2009168   1821882   
19  Month - 20   1305972   1559670   2107949   2151008   
20  Month - 21   1570152   2013615   1540016   2265814   
21  Month - 22   1481619   1527574   2255729   2295079   
22  Month - 23   1887694   1992155   1348387   1459683   
23  Month - 24   1880820   1752873   2098463   2052591   
24  Month - 25   1256333   1622671   1521792   1695122   
25  Month - 26   1422853   1715465   1853636   1520406   
26  Month - 27   1384426   1983163   1611169   1289160   
27  Month - 28   1616640   1547991   2128022   2178267   
28  Month - 29   1310654   1660092   1947119   1854412   

    Total sales per month of all zones  
0                              7022544  
1                              7152303  
2                              6475939  
3                              8174449  
4                              5995328  
5                              7151387  
6                              7287108  
7                              7816299  
8                              6703395  
9                              7128210  
10                             7032783  
11                             6111084  
12                             5925424  
13                             7155515  
14                             5934156  
15                             6506659  
16                             7149383  
17                             7083490  
18                             6971953  
19                             7124599  
20                             7389597  
21                             7560001  
22                             6687919  
23                             7784747  
24                             6095918  
25                             6512360  
26                             6267918  
27                             7470920  
28                             6772277

In [13]:
db["Total sales per month of all zones"].max()

8174449

In [14]:
db["Total sales per month of all zones"].min()

5925424

# 3. Check whether all the zones generate the same amount of sales.
H0 -μA = μB = μC = μD(where μ = mean)

Ha - At least one difference among the mean sales.

In [15]:
from scipy.stats import f_oneway
f_oneway(db['Zone - A'], db['Zone - B'], db['Zone - C'],db['Zone - D'])

F_onewayResult(statistic=5.672056106843581, pvalue=0.0011827601694503335)

# The F test statistic is 5.672056106843581 and the p-value is 0.0011827601694503335 then the p-value is less than .05, we reject the null hypothesis.Four zones has different sales.